knapiks/mcchat.py

```diff
@@ -104,80 +106,79 @@ def process_current_log():
+            player.last_login = msg.msg_time
+            player.save()
+
+            pending_messages.append(msg.id)
+        except Exception as e:
+            print(e)
+
+
+def update_logouts(result):
     # if no one is actually logged in, but the local db indicates someone may be logged in, process the current log
     # and mark them as logged out and send the sysop a message that they're logged out
+    players_without_logout_timestamps = get_active_players()
     if result == 'There are 0 of a max 20 players online: ' and len(players_without_logout_timestamps) > 0:
-        process_current_log()
         for player in players_without_logout_timestamps:
             last_logout = Log.objects.filter(msg_content=f'{player.name} left the game').last()
             player.last_logout = last_logout.msg_time
             player.save()
-            last_logout.msg_twilled = timezone.now()
-            last_logout.save()
-            sms = twilio_client.messages.create(
-                from_=f'+{twilio_phone}',
-                body=f'{player} logged out',
-                to=f'+{admin_phone}'
-            )
 
-    # if someone is logged in, process the log to  check for chats, logins and logouts--if any are found,
-    # send them to sysop in chronological order
-    if result != 'There are 0 of a max 20 players online: ':
-        process_current_log()
-        msg_list = []
-        # process logins first and update the player model
-        unsent_logins = Log.objects.filter(msg_content__contains='joined the game', msg_twilled=None)
-        for msg in unsent_logins:
-            player_name = re.match(player_pat, msg.msg_content).group()
-            try:
-                player = Player.objects.get_or_create(name=player_name)[0]
-                player.last_login = msg.msg_time
-                player.save()
-                msg_list.append(msg.id)
-            except Exception as e:
-                print(e)
-            msg.msg_twilled = timezone.now()
-            msg.save()
-
-        # then grab chats and logouts
-        unsent_chats = Log.objects.filter(msg_content__startswith='<', msg_twilled=None)
-        logouts = Log.objects.filter(msg_content__endswith='left the game', msg_twilled=None)
-        unsent_chats_and_logouts = unsent_chats | logouts
-        msgs_to_send = unsent_chats_and_logouts
-
-        for msg in msgs_to_send:
-            msg_list.append(msg.id)
-            msg.msg_twilled = timezone.now()
-            msg.save()
-
-        if len(msg_list) > 0:
-            messages_to_send = Log.objects.filter(id__in=msg_list)
+
+def send_messages:
+    if len(msg_list) > 0:
+            messages_to_send = Log.objects.filter(id__in=pending_messages).update(msg_twilled=timezone.now())
             messages = '\n'.join([msg_to_send.msg_content for msg_to_send in messages_to_send])
-            sms = twilio_client.messages.create(
+            twilio_client.messages.create(
```

Comment:
You were never doing anything with the sms variable, so you might as well not save it